### PR TITLE
Change methods of judging different material

### PR DESCRIPTION
**Pull request by FrankLeeX**

### CODE REVIEW

**Comment by FrankLeeX** [comment]
After carefully thinking about how we should determine two different materials where tunneling happens, I noticed that we always apply different source/drain voltage on different materials, which gives them different fLn and fLp. Thus I decides using this information as the judgment of different material. I didn't use the material name is that the number of layers in class Device and class ExtractData is different and we didn't have the parameter 'name' in ExtractData, so it can be hard and confusing to use 'name' in ExtractData where we have an array of semiconductors in the device. I think as long as we use different Drain/Source voltage, this method will always work fine.

File: Leno_beta1.5/src/Run1D.cpp [hunk 1]
```diff
@@ -118,13 +118,47 @@ void Run1D::runPoisson1D(int argc, char** argv){
...
+							//std::cout<<"Semi phin of ["<<i<<"] layer is:"<<data.phin[i]<<endl;
+							//std::cout<<"Semi phip of ["<<i<<"] layer is:"<<data.phip[i]<<endl;
+							//std::cout<<"Semi cBSemi of ["<<i<<"] layer is:"<<data.cBSemi[i]<<endl;
+							//std::cout<<"Semi vBSemi of ["<<i<<"] layer is:"<<data.vBSemi[i]<<endl;	
+						//	double judgecB = data.cBSemi[i]-data.cBSemi[i+1];
+						//	double judgevB = data.vBSemi[i]-data.vBSemi[i+1];
+						//	std::cout<<"judgecB = "<<judgecB<<endl;
+						//	std::cout<<"judgevB = "<<judgevB<<endl;
+						//	if ((abs(judgecB) > 0.5) || (abs(judgevB) > 0.5 )) {
+							if (data.fLnSemi[i] != data.fLnSemi[i+1]) {
```
Comment (Oscarlight) [inline Leno_beta1.5/src/Run1D.cpp]: What happens if VDS == 0? Will it goes to the last two layers? 
Also, you have assume the two layers are adjacent to each other. Will that be true? What if we have an insulator in between two layers?

**Comment by FrankLeeX** [comment]
This is the corrected version of the codes. I have tested it and it showed charge density correctly this time.

File: Leno_beta1.5/src/Run1D.cpp [hunk 2]
```diff
@@ -121,31 +121,7 @@ void Run1D::runPoisson1D(int argc, char** argv){
...
-							//std::cout<<"Semi cBSemi of ["<<i<<"] layer is:"<<data.cBSemi[i]<<endl;
-							//std::cout<<"Semi vBSemi of ["<<i<<"] layer is:"<<data.vBSemi[i]<<endl;	
-						//	double judgecB = data.cBSemi[i]-data.cBSemi[i+1];
-						//	double judgevB = data.vBSemi[i]-data.vBSemi[i+1];
-						//	std::cout<<"judgecB = "<<judgecB<<endl;
-						//	std::cout<<"judgevB = "<<judgevB<<endl;
-						//	if ((abs(judgecB) > 0.5) || (abs(judgevB) > 0.5 )) {
 							if (data.fLnSemi[i] != data.fLnSemi[i+1]) {
 								m=i;
 								n=i+1;
```
Comment (Oscarlight) [inline Leno_beta1.5/src/Run1D.cpp]: Add TODO here to remind the future user the flaky cases of this code 1) Vds == 0, 2) If semiconductor is used as the tunneling barrier.

File: Leno_beta1.5/src/Material.cpp [hunk 3]
```diff
@@ -71,21 +86,21 @@ double Material::electronDensity(double phin) {
 		//TODO: add Nta options in the input file. tried but ExtractData will not run correctly
 		// for DRC2016 Nta = 1600
 		// Feb 6th, 2017. Change 1600 to Nta. (By Frank)
-		if (Nta-0.0<1e-16) {
+		if (Nta-0.0 < 1e-16) {
```
Comment (Oscarlight) [inline Leno_beta1.5/src/Material.cpp:89]: I believe C++ double precision is 1e-16, so it is good enough to compare with 1e-14.

**Pull request closed without merging**